monai/data/utils.py

```diff
@@ -242,8 +242,24 @@ def worker_init_fn(worker_id: int) -> None:
 
     """
     worker_info = torch.utils.data.get_worker_info()
-    if hasattr(worker_info.dataset, "transform") and hasattr(worker_info.dataset.transform, "set_random_state"):
-        worker_info.dataset.transform.set_random_state(worker_info.seed % (2 ** 32))
+    set_rnd(worker_info.dataset, seed=worker_info.seed)
+
+
+def set_rnd(obj, seed: int) -> int:
+    """
+    Set seed or random state for all randomisable properties of obj.
+
+    Args:
+        seed: set the random state with an integer seed.
+    """
+    if not hasattr(obj, "__dict__"):
+        return seed  # no attribute
+    if hasattr(obj, "set_random_state"):
+        obj.set_random_state(seed=seed % (np.iinfo(np.int32).max + 1))
```

Comment:
If `seed` equals 2^31 then `seed % (np.iinfo(np.int32).max + 1)` would permit this value to be used which is one more than the maximum value for a 32-bit int (2^31-1). Did you perhaps mean `(seed + 1) % np.iinfo(np.int32).max`? If so before this line should be `seed = (seed + 1) % np.iinfo(np.int32).max` so that the correct in-bounds value is returned as well.

Reply:
yeah, I just realised some issues here as well. let me look into this again.
for now: 
- `np.iinfo(np.int32).max == (2**31 - 1)`
- `np.random.RandomState` requires seed between `0` and `2**32 - 1` (inclusive)
I hope it's the same across the different OS. there're a few places in the codebase need to be improved...